pandas/core/common.py

```diff
@@ -2439,6 +2439,8 @@ def _get_dtype_type(arr_or_dtype):
         return np.dtype(arr_or_dtype).type
     elif isinstance(arr_or_dtype, CategoricalDtype):
         return CategoricalDtypeType
+    elif arr_or_dtype is NotImplemented:
+        return type(NotImplemented)
```

Comment:
I get what you are saying about numpy types, but it shouldn't eat up any errors, since this function just says that the type of the input is so and so ... right?